compiler/rustc_lexer/src/lib.rs

```diff
@@ -522,8 +526,7 @@ impl Cursor<'_> {
         // Start is already eaten, eat the rest of identifier.
         self.eat_while(|c| {
             unicode_xid::UnicodeXID::is_xid_continue(c)
-                || (!c.is_ascii() && c.is_emoji_char())
-                || c == '\u{200d}'
+                || (!c.is_ascii() && c.is_emoji_char_or_emoji_component())
```

Comment:
Note that the '\u{200d}' here is already included as an emoji component, thus the removal.